spec/datadog/core/configuration/settings_spec.rb

```diff
@@ -497,39 +497,108 @@
         end
       end
 
-      describe '#allocation_counting_enabled' do
-        subject(:allocation_counting_enabled) { settings.profiling.advanced.allocation_counting_enabled }
```

Comment:
I think it may worth leaving a quick test in place, similar to what we have above:

```ruby
      describe '#allocation_counting_enabled=' do
        it 'logs a warning informing customers this no longer does anything' do
          expect(Datadog.logger).to receive(:warn).with(/no longer does anything/)
          settings.profiling.advanced.allocation_counting_enabled = true
        end
      end
```

In the spirit of, in the rare case that a customer is using this + we accidentally delete it, we'd still have a test that would prevent us from doing a backwards-incompatible change. 

(Of course for 2.0 we'll just rip-off the bandaid)